File: src/WorkflowShell/git/GitAmend.rb
require_relative '../models/Command'

class GitAmend < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'gitamend'
    @command_description = 'Amends the most recent commit to ensure that it has the correct author and an updated timestamp.'
    @command_usage = @command_string
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, true)

    command = "git commit --amend --reset-author --no-edit"
    run_shell_command(command, parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/git/GitAmend.ts
import Command from "../models/Command";

export default class GitAmend extends Command {

  static getString(): string {
    return "ga";
  }

  static getDescription(): string {
    return "Amends the most recent commit to ensure that it has the correct author and an updated timestamp.";
  }

  getUsage(): string {
    return GitAmend.getString();
  }

  async run(args: any) {
    const ok: boolean = await super.run(args);
    if (!ok) {
      return false;
    }

    const baseCommand: string = "git commit --amend --reset-author --no-edit";
    super.execute(baseCommand);
    return true;
  }
}
